python/redmonster/physics/zpicker2.py

```diff
@@ -138,11 +139,17 @@ def classify_obj(self, zfindobjs, zfitobjs, flags, rchi2threshold=0.005):
                             fiberminvecs.append(
                                     zfitobjs[itemp].minvectors[ifiber][imin])
                         except IndexError as e:
-                            print "%r" % e
-                            fibermins.append( \
-                                    n.max(zfitobjs[itemp].chi2vals[ifiber]) / \
-                                    (self.dof[ifiber] - zfindobjs[itemp].npoly))
+                            #print "%r" % e
+                            #fibermins.append( \
+                            #        n.max(zfitobjs[itemp].chi2vals[ifiber]) / \
+                            #        (self.dof[ifiber] - zfindobjs[itemp].npoly))
+
+                            # this is ok, might be no solution
+                            #print "WARNING no z fitted for fiber #%d, class #%d, zid #%d"%(ifiber,itemp,imin)
+
+                            fibermins.append(100000.)
```

Comment:
@julienguy changed that with the comment
> a bug has also been fixed when an IndexError is raised in the situation where no refined redshift is found for a given template class (I checked the best redshift/class from the other template classes is still return correctly).

If I interpret that correctly, the problem arises when `zfitobjs[itemp].chi2vals` is an empty list, which will cause `n.max(zfitobjs[itemp].chi2vals[ifiber])` to raise a `ValueError`.  The proposed fix of `fibermins.append(100000.)` also seems pretty hacky, but it is robust... :)  Thoughts on this case of no refined redshifts found for a given template class?